Content.Client/GameObjects/EntitySystems/ClientPhotoSystem.cs

```diff
@@ -0,0 +1,80 @@
+﻿using Content.Client.GameObjects.Components.Photography;
+using Content.Client.Interfaces.GameObjects;
+using Robust.Client.GameObjects.EntitySystems;
+using Robust.Client.Interfaces.Input;
+using Robust.Client.Player;
+using Robust.Shared.GameObjects.Systems;
+using Robust.Shared.Input;
+using Robust.Shared.Interfaces.Timing;
+using Robust.Shared.IoC;
+
+namespace Content.Client.GameObjects.EntitySystems
+{
+    public class ClientPhotoSystem : EntitySystem
+    {
+
+#pragma warning disable 649
+        [Dependency] private readonly IPlayerManager _playerManager = default;
+        [Dependency] private readonly IInputManager _inputManager = default;
+        [Dependency] private readonly IGameTiming _gameTiming = default;
+#pragma warning restore 649
+
+        private InputSystem _inputSystem;
+        private bool _blocked;
+        private bool _down;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            IoCManager.InjectDependencies(this);
+            _inputSystem = EntitySystemManager.GetEntitySystem<InputSystem>();
+        }
+
+        public override void Update(float frameTime)
```

Comment:
This seems like a strangely unnecessary approach to what should be a RangedAttack (or RangedInteract I believe it's called now). Was there a problem doing it that way?

Reply:
If you can point me towards the classes and files I'd be happy to rework it, as it stands this is the same approach guns do (Perhaps guns need the same refactor, or have some extra conditions that require this code)

Reply:
(I made a mistake, it's actually `IAfterInteract` that's necessary)

`IAfterInteract` on the server component should give click behaviour at range when no interaction happens, which could be used for the camera.
The downsides I can see are:
  1. There's a message shuffle - click goes to the server, then the server tells the client to take the photograph, then the client sends the photo back.
  2. Maybe there are some non-obvious behaviour differences?

The main reason I can think of that guns would use special code is that they are old and they need special handling for full-auto when you hold down the mouse. @PJB3005 You wrote it 2 years ago, any insights?

Reply:
@PJB3005 would you recommend leaving this as is for now then? (to avoid the issues you mentioned)

Reply:
You still don't need an update method. Now with chairbender's input binding rework you can just bind the Use key function going down instead of using an update loop like this.

Reply:
I thought the input rework wasn't merged yet? engine 1132?

Reply:
Ah no I see now, will do.